cli/src/commands/git.rs

```diff
@@ -1015,7 +1015,7 @@ fn cmd_git_push(
     .map_err(|err| match err {
         GitPushError::InternalGitError(err) => map_git_error(err),
         GitPushError::NotFastForward => user_error_with_hint(
-            "The push conflicts with changes made on the remote (it is not fast-forwardable).",
+            err.to_string(),
```

Comment:
nit: can be just `err` (to preserve the source error chain)

Reply:
Good point. I didn't know how this error chain magic works; I should look at it.

I'm actually thinking of refactoring this a bit, making this less relevant. I'd like to get rid of the "NotFastForward refactor" commit, put the new error rendering logic here, and thus either get rid of the `render_unexpected_ref_details` function (or whatever I called it) or move it into the CLI.

Reply:
I now did refactor this in a different direction (moved the logic to the CLI). I can undo or change it some more if you prefer.